### PR TITLE
allow setting hue of LedBar

### PR DESCRIPTION
**Pull request by lschloetterer**
This exposes the hue color functionality from the NeoPixel library.

I use it to set the CO2 color gradually like this:

```cpp
void setRGBledCO2color(int co2Value) {
  if (co2Value <= 0) return;
  co2Value = min(co2Value, 3000); // don't calculate hue beyond 3000 to not return green again
  uint16_t hue = map(co2Value, 500, 1800, UINT16_MAX/3 /*green*/, 0 /*red*/);
  ag.ledBar.setHue(hue);
}
```

### CODE REVIEW

File: src/Main/LedBar.cpp [hunk 1]
```diff
@@ -110,6 +110,17 @@ void LedBar::setColor(uint8_t red, uint8_t green, uint8_t blue) {
   }
 }
 
+/**
+ * @brief Sets the hue of all LEDs
+ *
+ * @param hue Hue of the color as 16-bit number
+ */
+void LedBar::setHue(uint16_t hue) {
```
Comment (nick-4711) [inline src/Main/LedBar.cpp:118]: Should we also support to set saturation and brightness, or is this not of interest?

File: src/Main/LedBar.cpp [hunk 2]
```diff
@@ -110,6 +110,17 @@ void LedBar::setColor(uint8_t red, uint8_t green, uint8_t blue) {
   }
 }
 
+/**
+ * @brief Sets the hue of all LEDs
```
Comment (nick-4711) [inline src/Main/LedBar.cpp:114]: I haven't worked with the HSV color model yet, but I would expect that we want to set this on a per LED base (and not for all).

Comment (nick-4711) [reply src/Main/LedBar.cpp:114]: @lschloetterer

Comment (lschloetterer) [reply src/Main/LedBar.cpp:114]: You're right, having the possibility to individually set the LEDs would be more dynamic.

However I see now that the code for the LED CO2 handler isn't in the arduino sketch anymore, but moved into the library in the AgStateMachine class. So having a custom handler isn't possible at the moment without modifying the library code anyway.

I'm not sure if there is a good solution to provide the possibility to have a custom LED handler function, I'd have to look into this when I have some more time

Comment (airgradienthq) [reply src/Main/LedBar.cpp:114]: @lschloetterer I am closing this PR for now. Please feel free to open a new PR at a later stage which is compatible with the current firmware.

**Pull request closed without merging**